### PR TITLE
Move to GitStaticAssets/4

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #32782

[V9 URL](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-9.0&branch=pr-en-us-33050)

DON"T USE AUTO GENERATED, it defaults to V8, not V9

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/static-files.md](https://github.com/dotnet/AspNetCore.Docs/blob/13da20cad21e156d8e3b98534f36b14bedb1813a/aspnetcore/fundamentals/static-files.md) | [Static files in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/static-files?branch=pr-en-us-33050) |

### CODE REVIEW

File: aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 app.UseHttpsRedirection();
 
-app.UseStaticFiles(new StaticFileOptions
+app.UseStaticFiles(new StaticFileOptions  //
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs:47]: Can't use `MapStaticAssets` here.

File: aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs [hunk 2]
```diff
@@ -383,7 +383,7 @@
 app.UseHttpsRedirection();
 
 // <snippet_mul>
-app.UseStaticFiles(); // Serve files from wwwroot
+app.MapStaticAssets(); // Serve files from wwwroot with MapStaticAssets
 app.UseStaticFiles(new StaticFileOptions
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs:387]: Probably shouldn't mix `MapStaticAssets();` with `UseStaticFiles`

**Review by Rick-Anderson — COMMENTED** [review]
@javiercn see my questions/remarks on when `MapStaticAssets` can replace `UseStaticFiles`

File: aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs (hunk 1 above)
Comment (javiercn) [reply aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs:47]: They are compatible, but depending on how `app.UseRouting` is used, one or the other will take precedence. For a simple sample like this, I would avoid updating it to use `app.MapStaticAssets` and probably create a separate section for it.

File: aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs [hunk 3]
```diff
@@ -383,18 +385,18 @@
...
-{
-    FileProvider = new PhysicalFileProvider(
-        Path.Combine(builder.Environment.ContentRootPath, "MyStaticFiles"))
-});
+ app.UseStaticFiles();
+ app.UseStaticFiles(new StaticFileOptions
+ {
+     FileProvider = new PhysicalFileProvider(
+         Path.Combine(builder.Environment.ContentRootPath, "MyStaticFiles"))
+ });
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/static-files/samples/9.x/StaticFilesSample/Program.cs:393]: @javiercn can you provide a `app.MapStaticAssets(staticAssetsManifestPath: "MyManifest");` and the format of the manifest file?
Take a look at all the `app.UseStaticFiles(new StaticFileOptions` occurances, I think this is the only one that can be replaced with `app.MapStaticAssets(staticAssetsManifestPath: "MyManifest");`

**Pull request merged**